### PR TITLE
Upgrade IngressClass to use v1 over v1Beta on Kube 1.19+

### PR DESCRIPTION
**Pull request by SantoDE**
### What does this PR do?

Updates the code to use the `IngressClass` Ressource with the apiVersion `networking/v1` if the Kubeversion is 1.19+

### Motivation

No more deprecation warnings

### More

- [x] Added/updated tests
- [ ] Added/updated documentation

### CODE REVIEW

File: pkg/provider/kubernetes/ingress/client.go [hunk 1]
```diff
@@ -465,24 +486,53 @@ func (c *clientWrapper) GetSecret(namespace, name string) (*corev1.Secret, bool,
...
+		return results, nil
+	}
+
 	if c.clusterFactory == nil {
 		return nil, errors.New("cluster factory not loaded")
 	}
 
+	isNetworkingV1Supported := supportsNetworkingV1Ingress(serverVersion)
+
+	if isNetworkingV1Supported {
```
Comment (dtomcej) [inline pkg/provider/kubernetes/ingress/client.go:504]:
```suggestion
	if  supportsNetworkingV1Ingress(serverVersion) {
```

**Review by jbdoumenjou — APPROVED** [review]
LGTM :+1:

**Review by dtomcej — APPROVED** [review]
LGTM
:shipit:

**Review by rtribotte — APPROVED** [review]
LGTM 👌

**Pull request merged**